src/format/parse.rs

```diff
@@ -219,7 +220,18 @@ fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseResult<(&'a st
         parsed.set_nanosecond(nanosecond)?;
     }
 
-    let offset = try_consume!(scan::timezone_offset_zulu(s, |s| scan::char(s, b':')));
+    let offset = if s.starts_with("UTC") || s.starts_with("utc") {
```

Comment:
1) The quote from RFC 3339 in the code comment above suggests the special string to search for would be `"Z"`. Why is that not the special case here?

<br/> 

2) Why not search the entire remainder of the string `s`? e.g. why not

    if matches!(s, "UTC" | "utc") {

   (also, `matches` is more succinct).  Would there be a case where `s` should have trailing data after having matched the timezone in an RFC3339?
   
It seems like this code expression should be:

    let offset = if matches!(s, "Z" | "z") {
        s = &[1..];
        0
    } else {
        ...
    }
    if ! s.empty() {
        return Err(TOO_LONG);
    }

Reply:
> 1. The quote from RFC 3339 in the code comment above suggests the special string to search for would be `"Z"`. Why is that not the special case here?

"Z" is handled by the parser. But this item does not only parse valid RFC 3339, but also supports "UTC" as offset. This is documented in https://docs.rs/chrono/latest/chrono/format/strftime/index.html#fn7. I suppose it makes some sense to support parsing this.
I'll add a comment.

> 2. Why not search the entire remainder of the string `s`? e.g. why not
>         if matches!(s, "UTC" | "utc") {
>         (also, `matches` is more succinct).  Would there be a case where `s` should have trailing data after having matched the timezone in an RFC3339?

I agree it is not likely to have any data after this.
But all parsing items only consume as much as they need to. There can be other items coming after it. And otherwise it is up to the calling function decide what to do with the trailing data.

Reply:
> "Z" is handled by the parser
Oh, it's handled by `scan::utc_offset(s, offset_format))` below. Gotcha.

> also supports "UTC" as offset. 
It is convenient but it's not strict RFC 3339, AFAICT. I'd prefer not allowing it. I won't block on this.

Reply:
> But all parsing items only consume as much as they need to.... And otherwise it is up to the calling function decide what to do with the trailing data.

Ah right, I recall now
https://github.com/chronotope/chrono/blob/c01e3a723883fc148c7e6052ddf4f6b42ff41e14/src/format/parse.rs#L519-L524

Reply:
> It is convenient but it's not strict RFC 3339, AFAICT. I'd prefer not allowing it.

The commit tile says what I think about it :smile::
> Remove hack to accept "UTC" in `timezone_offset_zulu`

- One place it is used is for the `%+` formatting item, with this description:
  > Same as `%Y-%m-%dT%H:%M:%S%.f%:z`, i.e. 0, 3, 6 or 9 fractional digits for seconds and colons in the time zone offset.
  >
  > This format also supports having a `Z` or `UTC` in place of `%:z`. They are equivalent to `+00:00`.
  >
  > Note that all `T`, `Z`, and `UTC` are parsed case-insensitively.
  >
  > The typical `strftime` implementations have different (and locale-dependent) formats for this specifier. While Chrono's format for `%+` is far more stable, it is best to avoid this specifier if you want to control the exact output.
  So being less strict here seems useful to me.

- `DateTime<FixedOffset>::parse_from_rfc_3339` also uses it, and should be very strict however.

- A second RFC3339 parser is in `impl str::FromStr for DateTime<FixedOffset>`.
  That one is very forgiving, which is useful there.
  Maybe parsing `%+` should use that implementation?